addons/im_livechat/static/tests/embed/livechat_session_tests.js

```diff
@@ -33,33 +34,62 @@ QUnit.test("Session is reset after failing to persist the channel", async (asser
     await contains(".o-mail-ChatWindow");
 });
 
-QUnit.test("Thread state is saved on the session", async (assert) => {
-    await startServer();
+QUnit.test("Fold state is saved on the server", async (assert) => {
+    const pyEnv = await startServer();
     await loadDefaultConfig();
     const env = await start();
     await click(".o-livechat-LivechatButton");
     await contains(".o-mail-Thread");
-    assert.strictEqual(env.services["im_livechat.livechat"].sessionCookie.state, "open");
+    await insertText(".o-mail-Composer-input", "Hello World!");
+    triggerHotkey("Enter");
+    await contains(".o-mail-Message", { text: "Hello World!" });
+    let [member] = pyEnv["discuss.channel.member"].searchRead([
+        ["guest_id", "=", pyEnv.currentGuest.id],
+        ["channel_id", "=", env.services["im_livechat.livechat"].thread.id],
+    ]);
+    assert.strictEqual(member.fold_state, "open");
     await click(".o-mail-ChatWindow-header");
     await contains(".o-mail-Thread", { count: 0 });
-    assert.strictEqual(env.services["im_livechat.livechat"].sessionCookie.state, "folded");
+    [member] = pyEnv["discuss.channel.member"].searchRead([
+        ["guest_id", "=", pyEnv.currentGuest.id],
+        ["channel_id", "=", env.services["im_livechat.livechat"].thread.id],
+    ]);
+    assert.strictEqual(member.fold_state, "folded");
     await click(".o-mail-ChatWindow-header");
-    await contains(".o-mail-Thread");
-    assert.strictEqual(env.services["im_livechat.livechat"].sessionCookie.state, "open");
 });
 
-QUnit.test("Seen message is saved on the session", async (assert) => {
-    await startServer();
+QUnit.test("Seen message is saved on the server", async (assert) => {
+    const pyEnv = await startServer();
     await loadDefaultConfig();
     const env = await start();
     await click(".o-livechat-LivechatButton");
-    assert.notOk(env.services["im_livechat.livechat"].sessionCookie.seen_message_id);
-    await insertText(".o-mail-Composer-input", "Hello World!");
+    await contains(".o-mail-Thread");
+    await insertText(".o-mail-Composer-input", "Hello, I need help!");
     triggerHotkey("Enter");
-    await contains(".o-mail-Message", { count: 2 });
-    await nextTick(); // wait for message seen
+    await contains(".o-mail-Message", { text: "Hello, I need help!" });
+    const initialSeenMessageId = env.services["im_livechat.livechat"].thread.seen_message_id;
+    await pyEnv.withUser(pyEnv.adminUserId, () =>
+        rpc("/mail/message/post", {
+            post_data: {
+                body: "Hello World!",
+                message_type: "comment",
+                subtype_xmlid: "mail.mt_comment",
+            },
+            thread_id: env.services["im_livechat.livechat"].thread.id,
+            thread_model: "discuss.channel",
+        })
+    );
+    $(".o-mail-Composer-input").blur();
+    await contains(".o-mail-Thread-newMessage");
+    await contains(".o-mail-Composer-input", { setFocus: true });
```

Comment:
Looks safer to `.blur()` before `await /mail/message/post`, as this could potentially already render the message and not reach the expected out-of-focus "new message" received flow